app/controllers/api/v1/admin/roles_controller.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+module Api
+  module V1
+    module Admin
+      class RolesController < ApiController
+        skip_before_action :verify_authenticity_token
+        before_action -> { config_sorting(allowed_columns: %w[name]) }, only: :index, if: -> { params.key?(:sort) }
```

Comment:
I'm not a huge fan of this style of programming - I find it extremely difficult to follow. I'd rather we manually call it in the action and return values rather than setting instance variables through before_actions

Reply:
The problem is with the difference between a regular method and a callback.
Regular methods if called the `render|redirect_to` will not stop the request response cycle which will lead us to do some extra logic in order to ensure the expected completion of the response like passing a reference or returning certain outputs.
Do you approve this style instead:
```
before_action do
  # Code
end
```

Reply:
I would just call the method directly and ignore anything that isn't allowed. There's no need for that method to raise an error